### ISSUE TITLE
Redirect on custom plugin

### ISSUE THREAD

**Issue by joscha** [pecuniabanking/pecunia-client#429]
I tried to implement a plugin for [NAB](http://nab.com.au), however it seems as if the base URL (https://ib.nab.com.au/nabib/index.jsp) gets redirected to `about:blank` in some strange way. When using a "normal" browser or curl, it seems to work. The debug output is:

```console
[Verbose] (*) Start loading
[Verbose] (*) Closing frame...
[Verbose] (*) Committed load for frame
[Verbose] (*) JS create
[Verbose] (*) Performing client redirection...
[Verbose] (*) Finished loading frame from URL: https://ib.nab.com.au/nabib/index.jsp
[Verbose] (*) Start loading
[Verbose] (*) Closing frame...
[Verbose] (*) Committed load for frame
[Verbose] (*) JS create
[Verbose] (*) Finished loading frame from URL: about:blank
[Verbose] (*) Calling callback...
```

Any ideas on what I might be doing wrong?

**Comment by mike-lischke** [comment]
Try working with the built-in browser window (klick the Safari icon). You can even work there manually and see what the navigation looks like. Just the initial URL must be loaded in the script.

**Comment by joscha** [comment]
Hi @mike-lischke - thank you. Actually, when I try and use the builtin browser, the same thing happens. The page does not open and `about:blank` is loaded. Any other idea?

**Comment by joscha** [comment]
I guess I found the culprit:
```
isSafari = (navigator.appVersion.indexOf("Safari")!=-1);
if (!isSafari)
{
 if (window.name == "") {window.location="about:blank"}
}
```

**Comment by mike-lischke** [comment]
But shouldn't Webkit identify itself as Safari? Maybe you can just set an arbitrary window name?

**Comment by joscha** [comment]
I tried
`webClient.customUserAgent = "Pecunia/1.2.1";` in `PluginWorker.swift` but it doesn't seem to work. I think there is some other form of user agent sniffing on that website.

**Comment by joscha** [comment]
Adding a
`webClient.applicationNameForUserAgent = "Pecunia/1.2.1 (Safari)";` works. Basically the browser sniffing is broken for the banking website. Any possibility to add an interface for `applicationNameForUserAgent` or [`userAgentForURL`](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/WebKit/Classes/WebView_Class/#//apple_ref/occ/instm/WebView/userAgentForURL) to the PluginWorker?

**Comment by mike-lischke** [comment]
Why an interface? Can't we just set this user agent in swift and be done with it it?

**Comment by joscha** [comment]
Yes, we definitely could, but maybe a different banking site has a
different (weird) way of sniffing the user agent, so I thought that
switching the user agent to `Pecunia/x.x.x (Safari)` might break  someone
else's plugin.

**Comment by joscha** [comment]
Closing this for now, as adding a

```javascript
webClient.applicationNameForUserAgent = "Pecunia/1.2.1 (Safari)";
```
to the `getStatements` function on JavaScript side actually does the trick.

**Comment by joscha** [comment]
implemented in 9255a7bd9bbbe54677678fd88c0b1fa2c956aeb2
